### PR TITLE
Predict general interactions.

### PR DESCRIPTION
**Pull request by ElectroJr**
The main change that this PR makes is to move the interaction keybinding handling from server-side to shared code. It also moves `UseDelaySystem` to shared and makes `UseDelayComponent` networked. 

These changes should mean that any interactions defined in shared code should start getting predicted (even if they weren't written with that in mind). The main noticeable benefit is that item pickup interactions are predicted, so there is no longer an odd delay after clicking on an item.

This PR is likely to introduce some bugs, e.g:
- Double audio, pop-ups, and animations
- mispredicts if there are both shared & server-exclusive interactions available, and the server performs the server-side one while the client runs some other shared interaction.
- reconciliation bugs

IMO its worth just making the change and fixing the issues as they come to light. If they are too severe this can just be reverted.

I haven't tested all that thoroughly so far, but haven't noticed anything. Though beware that my headset appears to be dead at the moment, so I have no idea what the audio is up to.

:cl:
- tweak: The client now tries to predict more interactions. This may lead to some bugs, particularly double audio, pop-ups, and animations. Please report any that come up.

### CODE REVIEW

File: Content.Shared/Timing/UseDelaySystem.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+            _activeDelays.Remove(component);
+        }
+        else
+
+        if (component.RemainingDelay == null)
+            return;
+
+        component.DelayEndTime = _gameTiming.CurTime + component.RemainingDelay;
+        Dirty(component);
+        _activeDelays.Add(component);
```
Comment (metalgearsloth) [inline Content.Shared/Timing/UseDelaySystem.cs:41]: My indentation

**Review by metalgearsloth — APPROVED** [review]
(I assume it's okay ingame)

**Comment by metalgearsloth** [comment]
what if I just merged

**Review by ElectroJr — COMMENTED** [review]
> what if I just merged

Welp I'm gonna be offline in an hour or so, so if a barrage of bug reports do come in I won't be round to help.

Though when checking just now I already found a double audio issue with clicking to open doors, so imma fix that quickly

**Pull request merged**